controllers/admin/self-managed/UpdatePageUpdateOptionsController.php

```diff
@@ -49,23 +54,81 @@ protected function displayRouteInUrl(): ?string
         return Routes::UPDATE_PAGE_UPDATE_OPTIONS;
     }
 
+    public function saveOption(): JsonResponse
+    {
+        $upgradeConfiguration = $this->upgradeContainer->getUpgradeConfiguration();
+        $upgradeConfigurationStorage = $this->upgradeContainer->getUpgradeConfigurationStorage();
+
+        $name = $this->request->request->get('name');
+
+        $config = [
+            $name => $this->request->request->get('value'),
+        ];
+
+        $error = $this->upgradeContainer->getConfigurationValidator()->validate($config);
```

Comment:
Data structure is incorrect for template and needs to be reworked before being sent. Reuse logic from `controllers/admin/self-managed/UpdatePageVersionChoiceController.php::save()`